doc/semiboolmat.xml

```diff
@@ -0,0 +1,150 @@
+##
+
+<#GAPDoc Label="RegularBooleanMatMonoid">
+<ManSection>
+  <Oper Name="RegularBooleanMatMonoid" Arg="d"/>
+  <Returns>A monoid of boolean matrices.</Returns>
+  <Description>
+    If <A>d</A> is a positive integer, then
+    <C>RegularBinaryRelationSemigroup</C> returns the monoid generated by the
+    regular <A>d</A> by <A>d</A> boolean matrices. Note that this monoid is
+    <E>not</E> regular in general.  <C>RegularBooleanMatMonoid(<A>d</A>)</C> is
+    generated by the four boolean matrices <C>A, B, C, D</C>, whose <K>true</K>
+    entries are:
+    <List>
+      <Item>
+        <C>A[i][i + 1]</C> and <C>A[n][1]</C>, for
+        <M>i \in \{1, \ldots, n - 1\}</M>;
+      </Item>
+      <Item>
+        <C>B[1][2]</C>, <C>B[2][1]</C>, and <C>B[i][i]</C> for <M>i \in \{3,
+          \ldots, n\}</M>;
+      </Item>
+      <Item>
+        <C>C[1][2]</C> and <C>C[i][i]</C>, for <M>i \in \{2, \ldots, n -
+          1\}</M>; and
+      </Item>
+      <Item>
+        <C>D[1][2]</C>, <C>D[i][i]</C>, for <M>i \in \{2, \ldots, n\}</M>,
+        and <C>D[n][1]</C>.
+      </Item>
+    </List>
+    This monoid has nearly <C>2 ^ (n ^ 2)</C> elements. 
+  </Description>
+</ManSection>
+<#/GAPDoc>
+
+<#GAPDoc Label="FullBooleanMatMonoid">
+   <ManSection>
+    <Oper Name="FullBooleanMatMonoid" Arg="d"/>
+    <Returns>A monoid of boolean matrices.</Returns>
+    <Description>
+      If <A>d</A> is a positive integer, then this operation returns full
+      boolean matrix monoid, which is the monoid consisting of all <A>d</A> by
+      <A>d</A> boolean matrices. This monoid consists of <C>2 ^ (<A>n</A> ^
+        2)</C> elements.
+
+      <Example><![CDATA[
+gap> S := FullBooleanMatMonoid(3);
+<monoid of 3x3 boolean matrices with 5 generators>
+gap> Size(S);
+512]]></Example>
+    </Description>
+  </ManSection>
+<#/GAPDoc>
+
+<#GAPDoc Label="HallMonoid">
+   <ManSection>
+    <Oper Name="HallMonoid" Arg="d"/>
+    <Returns>A monoid of boolean matrices.</Returns>
+    <Description>
+      If <A>d</A> is a positive integer, then this operation returns the monoid
+      consisting of all <A>d</A> by <A>d</A> hall matrices. A <E>hall matrix</E>
+      is a boolean matrix in which every column and every row contains at least
+      one <K>true</K> entry. Equivalently, a hall matrix is a boolean matrix
+      than contains a permutation.
+
+      <Example><![CDATA[
+gap> S := HallMonoid(3);
+<monoid of 3x3 boolean matrices with 4 generators>
+gap> Size(S);
+247]]></Example>
+    </Description>
+  </ManSection>
+<#/GAPDoc>
+
+<#GAPDoc Label="GossipMonoid">
+   <ManSection>
+    <Oper Name="GossipMonoid" Arg="d"/>
+    <Returns>A monoid of boolean matrices.</Returns>
+    <Description>
+      If <A>d</A> is a positive integer, then this operation returns the
+      <A>d</A> by <A>d</A> gossip monoid. The <E>gossip monoid</E> is the
+      monoid generated by the collection of all <A>d</A> by <A>d</A> boolean
+      matrices that define an equivalence relation.
```

Comment:
It'd be good to have a link to one or two of the papers about this. The size of the Gossip monoid is not known for n greater than some relatively small value. This paper https://arxiv.org/abs/1405.5979 in particular, where they show that the size of the gossip monoid is whatever it is, up to n = 9, see page 20.